sched/distributor/poolDistributor_test.go

```diff
@@ -0,0 +1,68 @@
+package distributor
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	"testing"
+)
+
+func TestPoolDistributor_NilCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nil)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_EmptyCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := make([]cm.Node, 0)
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nodes)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_OneNodeCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := cm.GenerateTestNodes(1)
+	clusterMock := cm.NewMockCluster(mockCtrl)
```

Comment:
fwiw I think we should be judicious with our use of mocks. They're ok to have but I wouldn't want them to be our go-to-test-type.